ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx

```diff
@@ -82,12 +71,39 @@ const WalletPaymentPickPspScreen = () => {
     supportRequest: true
   });
 
+  useHeaderSecondLevel({
+    title: "",
+    contextualHelp: emptyContextualHelp,
+    faqCategories: ["payment"],
+    supportRequest: true
+  });
+
+  const walletId = pipe(
+    selectedWalletOption as O.Option<WalletInfo>,
+    O.fold(
+      () => "",
+      (wallet: WalletInfo) => wallet.walletId
+    )
+  );
+
   useFocusEffect(
     React.useCallback(() => {
-      dispatch(
-        walletPaymentCalculateFees.request({ walletId, paymentAmountInCents })
+      pipe(
+        paymentAmountPot,
+        pot.toOption,
+        O.fold(
+          () => null, // should display error alert
+          paymentAmountInCents => {
+            dispatch(
+              walletPaymentCalculateFees.request({
+                walletId,
+                paymentAmountInCents
+              })
+            );
+          }
+        )
```

Comment:
It could be simplified. What do you think?

```suggestion
     pipe(
        sequenceT(O.Monad)(
          pot.toOption(paymentAmountPot),
          selectedWalletOption
        ),
        O.map(([paymentAmountInCents, selectedWallet]) => {
          dispatch(
            walletPaymentCalculateFees.request({
              walletId: selectedWallet.walletId,
              paymentAmountInCents
            })
          );
        })
      );
```